### PR TITLE
Adding conditionally as a fix for order of cond

### LINKED ISSUE #98: Either.cond parameter order mismatch
All of our APIs have errors on the left...except for that one.

https://github.com/arrow-kt/arrow-core/blob/master/arrow-core-data/src/main/kotlin/arrow/core/Either.kt#L866

As seen in https://youtu.be/tN_k1W1hEhA?t=7094

### PR DESCRIPTION
**Pull request by sierisimo**
This commit resolves #98 with the new function `conditionally` as
proposed in Slack by @hiosdra

It also moves `cond` to just pass it's parameters to `conditionally`

### CODE REVIEW

**Comment by Hiosdra** [comment]
Great 👍 
One thing ->  Shoudn't `Either.cond` be made deprecated in favour of new method if we move to `conditionally` in tests?

**Comment by sierisimo** [comment]
I wanted baby steps on moving out `cond` but I can add the `@Deprecated(…)`

Or should we just kill it with fire? 

WITH FIRE I SAID

**Comment by pakoito** [comment]
Let's be nice and deprecate first :D

File: arrow-core-data/src/main/kotlin/arrow/core/Either.kt [hunk 1]
```diff
@@ -416,14 +416,14 @@ import arrow.typeclasses.Show
  * }
  * ```
  *
- * For creating Either instance based on a predicate, use `Either.cond()` method :
+ * For creating Either instance based on a predicate, use `Either.conditionally()` method :
```
Comment (pakoito) [inline arrow-core-data/src/main/kotlin/arrow/core/Either.kt:419]: could you please add something about parameter order here?

File: arrow-core-data/src/main/kotlin/arrow/core/Either.kt [hunk 2]
```diff
@@ -863,7 +863,9 @@ sealed class Either<out A, out B> : EitherOf<A, B> {
       }
     }
 
-    fun <L, R> cond(test: Boolean, ifTrue: () -> R, ifFalse: () -> L): Either<L, R> = if (test) right(ifTrue()) else left(ifFalse())
+    fun <L, R> cond(test: Boolean, ifTrue: () -> R, ifFalse: () -> L): Either<L, R> = conditionally(test, ifFalse, ifTrue)
+
+    fun <L, R> conditionally(test: Boolean, ifFalse: () -> L, ifTrue: () -> R): Either<L, R> = if (test) right(ifTrue()) else left(ifFalse())
```
Comment (pakoito) [inline arrow-core-data/src/main/kotlin/arrow/core/Either.kt:868]: Some docs atop this would be nice. Not a requierement for merging, just nice.

**Comment by sierisimo** [comment]
Docs added to explain how this works!

File: arrow-core-data/src/main/kotlin/arrow/core/Either.kt [hunk 3]
```diff
@@ -863,7 +865,28 @@ sealed class Either<out A, out B> : EitherOf<A, B> {
       }
     }
 
-    fun <L, R> cond(test: Boolean, ifTrue: () -> R, ifFalse: () -> L): Either<L, R> = if (test) right(ifTrue()) else left(ifFalse())
+    @Deprecated(
+      message = "use conditionally as parameters are consistent with Either class",
```
Comment (pakoito) [inline arrow-core-data/src/main/kotlin/arrow/core/Either.kt:869]: *parameter order

**Comment by pakoito** [comment]
To prod!

**Pull request merged**